### PR TITLE
CORE-249 Poetry Upgrade 

### PR DESCRIPTION
**Pull request by stuart-bradley**
### Jira-ID: [CORE-249](https://yoyowallet.atlassian.net/browse/CORE-249)

This PR does a number of things:
- [x] Removes support for Python 3.6 (it's 5 years old).
- [x] Adds support for Python 3.9 and 3.10.
- [x] Moves dependency management to Poetry.
- [x] Moves CI to Github Actions. 
- [x] Upgrades dependencies.
- [x] Fixes failing tests. 

I'll move this Repo to PyPi in a seperate PR using our dev@yoyowallet.com account.    

Once this is merged in I'll remove CircleCI completely.

### CODE REVIEW

File: .github/actions/drf-integrations-framework-build/action.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      with:
+        path: .venv
+        key: venv-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+    - name: Install dependencies
+      if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+      run: poetry install --no-interaction --no-root
+      shell: bash
+    - name: Install Django version
+      run: python -m pip install "Django~=${{ inputs.django-version }}"
+      shell: bash
```
Comment (stuart-bradley) [inline .github/actions/drf-integrations-framework-build/action.yml:37]: Copied from https://github.com/yoyowallet/drf-react-template-framework/blob/master/.github/actions/drf-react-template-framework-build/action.yml - will move to a separate repo in the near future.

**Comment by stuart-bradley** [comment]
> Nice! Should we also update the changelog and the version in `drf_integrations/__init__.py`?

Are there any actual changes for end users?

**Comment by delenamalan** [comment]
> Are there any actual changes for end users? Also why are we keeping the Changelog updated? Shouldn't the release logs perform that function?

I would put the Python version requirement update in the Changelog at least.

With release logs, do you mean the Github releases? I think we've been considering the Changelog the source of truth for changes. If we rather want to use Github releases for that purpose instead, we should remove the Changelog.

**Comment by stuart-bradley** [comment]
> With release logs, do you mean the Github releases? I think we've been considering the Changelog the source of truth for changes. If we rather want to use Github releases for that purpose instead, we should remove the Changelog.

Yeah, I'll remove the changlog and add the removal of Python 3.6 when I make a release (which I won't do with this PR)/

**Review by delenamalan — APPROVED** [review]

File: .github/actions/drf-integrations-framework-build/action.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      id: cached-poetry-dependencies
+      uses: actions/cache@v2
+      with:
+        path: .venv
+        key: venv-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+    - name: Install dependencies
+      if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+      run: poetry install --no-interaction --no-root
+      shell: bash
+    - name: Install Django version
```
Comment (crintus) [inline .github/actions/drf-integrations-framework-build/action.yml:35]: Why do you still have to pip install django?

**Review by crintus — APPROVED** [review]

File: .github/actions/drf-integrations-framework-build/action.yml (hunk 2 above)
Comment (stuart-bradley) [reply .github/actions/drf-integrations-framework-build/action.yml:35]: Because it installs a different version to what's in the `.lock` file for testing.

**Pull request merged**